connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java

```diff
@@ -98,20 +105,21 @@ public List<InboundConnectorDefinitionImpl> findInboundConnectors(
 
   private List<InboundConnectorDefinitionImpl> inspectBpmnProcess(
       Process process, ProcessDefinition definition) {
-
-    List<BaseElement> inboundEligibleElements =
-        INBOUND_ELIGIBLE_TYPES.stream()
-            .flatMap(type -> process.getChildElementsByType(type).stream())
-            .filter(
-                element -> {
-                  Map<String, String> zeebeProperties = getRawProperties(element);
-                  if (zeebeProperties != null
-                      && zeebeProperties.get(Keywords.INBOUND_TYPE_KEYWORD) != null) {
-                    return true;
-                  }
-                  return false;
-                })
-            .collect(Collectors.toList());
+    Set<FlowElement> allElements = retrieveEligibleElementsFromProcess(process);
+    Set<BaseElement> inboundEligibleElements = new HashSet<>();
+    for (FlowElement element : allElements) {
+      for (Class clz : INBOUND_ELIGIBLE_TYPES) {
+        if (clz.isInstance(element)) {
+          // this FlowElement type currently does not support
+          // inbound connectors implementations
+          continue;
+        }
+        final var zeebeProperties = getRawProperties(element);
+        if (zeebeProperties != null && zeebeProperties.get(INBOUND_TYPE_KEYWORD) != null) {
+          inboundEligibleElements.add(element);
+        }
+      }
```

Comment:
Had to think so some seconds about what is happening here, could we simplify it a little bit?

`INBOUND_ELIGIBLE_TYPES` can be a Set<Class>

```
if(INBOUND_ELIGIBLE_TYPES.contains(element.getClass()) && ...) {
  ...
}
```

Lets also move this method into `retrieveEligibleElementsFromProcess` and filter there. I would assume that a method called `retrieveEligibleElementsFromProcess` only returns eligibile lements.

Reply:
Sure, can rework. Please note, that `INBOUND_ELIGIBLE_TYPES.contains(element.getClass())` won't work because, for example, for start events, the `INBOUND_ELIGIBLE_TYPES` contain `StartEvent.class` interface, while the actual result of `element.getClass()` will be `class io.camunda.zeebe.model.bpmn.impl.instance.StartEventImpl` thus will be always `false`.

Reply:
Done

Reply:
@igpetrov Thank you! Maybe we should rely on `ModelElementType.getInstanceType()` or `ModelElementType.getTypeName()` for a more robust and simpler approach in the future.